src/requests/create-signature.ts

```diff
@@ -0,0 +1,120 @@
+import * as crypto from 'crypto'
+import type {
+  CanonicalRequest,
+  NormalizedCanonicalRequest,
+  NormalizedHeaders,
+  Secret,
+  Timestamp,
+} from './typings'
+import { CanonicalRequestValidator, SecretValidator, TimestampValidator } from './typings'
+import { getNormalizedEncodedURI, getNormalizedHeaders } from './utils'
+import { CONTENTFUL_HEADERS, ContentfulHeader } from './constants'
+
+const hash = (normalizedCanonicalRequest: NormalizedCanonicalRequest, secret: string) => {
+  const stringifiedHeaders = normalizedCanonicalRequest
+    .headers!.map(([key, value]) => `${key}:${value}`)
+    .join(';')
+
+  const stringifiedRequest = [
+    normalizedCanonicalRequest.method,
+    normalizedCanonicalRequest.path,
+    stringifiedHeaders,
+    normalizedCanonicalRequest.body,
+  ].join('\n')
+
+  const hmac = crypto.createHmac('sha256', secret)
+
+  hmac.update(stringifiedRequest)
+
+  return hmac.digest('hex')
+}
+
+const removeExistingContentfulHeaders = (headers: NormalizedHeaders) => {
+  return headers.filter(([key]) => !CONTENTFUL_HEADERS.includes(key))
+}
+
+const enrichNormalizedHeadersWithMetadata = (headers: NormalizedHeaders, timestamp: number) => {
+  const result = removeExistingContentfulHeaders(headers)
+  const joinedSignedHeaders = result
+    .map(([key]) => key)
+    // We always sign timestamp and signed headers
+    .concat(ContentfulHeader.Timestamp, ContentfulHeader.SignedHeaders)
+    .join(',')
```

Comment:
let's sort before joining. Remember it will be 2 different environments doing signing, they may even not use JS